src/actions/sources/newSources.js

```diff
@@ -46,20 +47,24 @@ function createOriginalSource(
   };
 }
 
-function loadSourceMaps(sources) {
-  return async function({ dispatch, sourceMaps }: ThunkArgs) {
+function loadSourceMaps(sources: Source[]) {
+  return async function({
+    dispatch,
+    sourceMaps
+  }: ThunkArgs): Promise<Promise<Source>[]> {
     if (!prefs.clientSourceMapsEnabled) {
-      return;
+      return [];
     }
 
-    let originalSources = await Promise.all(
-      sources.map(({ id }) => dispatch(loadSourceMap(id)))
+    return flatten(
+      sources.map(async ({ id }) => {
+        const originalSources = await dispatch(loadSourceMap(id));
+        if (originalSources !== undefined) {
```

Comment:
I think now we don't need this check, since we know it'd never be `undefined`

Reply:
nope, no objection.

Reply:
Alright, I've annotated `loadSourceMap` to return `Promise<Source[]>` and fixed the return points. I've updated `test-head.js` so that all tests initialize the `sourceQueue` as per my previous comment.

Then I've actually reinstated the following check tho:
```diff
+        const originalSources = await dispatch(loadSourceMap(id));
+        if (originalSources.length > 0) {
+          sourceQueue.queueSources(originalSources);
+        }
```

Otherwise whenever called - even with an empty array - `queueSources` internally will create a new array and schedule a new dispatchNewSources call, which seems not ideal. I opted for checking that the array is not empty before calling `queueSources` since it's closer to where the array is created, and it's obvious why it'd be valid for it to be empty. You could argue that `queueSources` should return early if `sources.length == 0` tho and I'm happy to do that if anyone thinks it's better.